lib/modules/xPostLinks.js

```diff
@@ -0,0 +1,61 @@
+addModule('xPostLinks', function(module, moduleID) {
+	module.moduleName = 'X-post Links';
+	module.category = 'Posts';
+	module.description = 'Create links to x-posted subreddits in the taglines.';
+	module.include = [ 'linklist', 'comments', 'profile' ];
+
+	module.go = function() {
+		if ((module.isEnabled()) && (module.isMatchURL())) {
+
+			createLinks();
+			RESUtils.watchForElement('siteTable', createLinks);
+		}
+	};
+
+	function createLinks( container ) {
+		var
+		titles        = (container || document.body).querySelectorAll('a.title'),
+		xpost_re      = /x-?post/i,
+		xpost_from_re = /(x-?post\w*\b)(\s+\w+)?\s+([\/r]*)?([a-zA-Z0-9_]+)/i, // get stuff like x-post from games, x-post games, xPost /games ... this is the fallback
+		subreddit_re  = /r\/([a-zA-Z0-9_]+)/i; // get stuff like x-post from a neat subreddit called r/games
+
+		for (var i = 0, title, tagline, entry, userattrs, textNode, subreddit, add_after, xpost_from, xpost_from_a, sub; i < titles.length; i++) {
+
+			title = titles.item(i);
+
+			if ( xpost_re.test(title.textContent) ){
+
+				sub        = title.textContent.match(subreddit_re);
+				xpost_from = title.textContent.match(xpost_from_re);
+
+				sub        = sub && sub.length >= 1 && sub[1];
+				xpost_from = xpost_from && xpost_from.length >= 5 && xpost_from[4];
+
+				if ( sub || xpost_from ){
+
+					sub = sub || xpost_from;
```

Comment:
I've been aware of the problem you mention, however the other way creates other problem cases. I've put some more thought into it and came to this conclusion:
I think we can agree that if after "x-post" something with the pattern of r/string is found, we can be pretty sure we found gold. That's priority 1.
Suggested priority 2: if we find one or two words after the "x-post" part we can be pretty sure it's either something like " from askReddit" or just the subreddit. 
Problemematic are cases with more than 2 words after the x-post. We could guess the subreddit is probably the last word, but i've already encountered a case where this would fail. I'd say we just skip those. Should be pretty rare.
I made some tests which showed prio 1 would be 80-90% of all cases, prio 2 the rest. In about 200 x-post finds none failed (only tested the regex programatically, no manual evaluation)

try yourself
https://www.reddit.com/search?q=x-post+subreddit%3Abooks&restrict_sr=&sort=top&t=all

    var el = document.querySelectorAll(".search-title");
    var xpost_re      = /(x-?post\S*)(.+)/i;
    var xpost_from_re = /^(\s+\S+)?\s+\/?([a-zA-Z0-9_]+)([\)\]}])|($)/i;
    var subreddit_re  = /r\/([a-zA-Z0-9_]+)/i; 
    var s;
    var success = {
        "xpost_re": 0,
        "xpost_from_re": 0,
        "subreddit_re": 0
    };
    for (var i = 0; i < el.length; i++) {
        
        if ( el[i].textContent.match(xpost_re) ){
            success["xpost_re"]++;
            s = el[i].textContent.match(xpost_re)[2];
            if ( s.match(subreddit_re) ){
                success["subreddit_re"]++;
            } 
            else if ( s.match(xpost_from_re) ){
                success["xpost_from_re"]++;
            }
        } 
    }
    console.log(success);

will commit later